src/main/scala/no/ndla/frontpageapi/service/WriteService.scala

```diff
@@ -19,26 +19,55 @@ trait WriteService {
 
   class WriteService {
 
-    def newSubjectPage(subject: api.NewOrUpdateSubjectFrontPageData): Try[api.SubjectPageData] = {
+    def newSubjectPage(subject: api.NewSubjectFrontPageData): Try[api.SubjectPageData] = {
       for {
         convertedSubject <- ConverterService.toDomainSubjectPage(subject)
-        subjectPage <- subjectPageRepository.newSubjectPage(convertedSubject, subject.externalId)
-      } yield ConverterService.toApiSubjectPage(subjectPage, "nb")
+        subjectPage <- subjectPageRepository.newSubjectPage(convertedSubject, subject.externalId.getOrElse(""))
+        converted <- ConverterService.toApiSubjectPage(subjectPage, "nb")
+      } yield converted
     }
 
-    def updateSubjectPage(id: Long, subject: api.NewOrUpdateSubjectFrontPageData): Try[api.SubjectPageData] = {
+    def updateSubjectPage(id: Long,
+                          subject: api.NewSubjectFrontPageData,
+                          language: Option[String]): Try[api.SubjectPageData] = {
```

Comment:
Ser ikke ut som `language` er brukt i funksjonen?
Stemmer også for å gjøre det om til en vanlig `String` og heller håndtere defaults i controlleren sånn som vi gjør andre steder :)